clang/lib/Sema/SemaExpr.cpp

```diff
@@ -4903,6 +4930,21 @@ Sema::CreateUnaryExprOrTypeTraitExpr(Expr *E, SourceLocation OpLoc,
     E = PE.get();
   }
 
+  if (ExprKind == UETT_NameOf) {
+    EnumNameVisitor obj;
+    obj.VisitExpr(E);
+    StringRef EnumName = obj.getEnumName();
+    QualType StrTy =
+      Context.getStringLiteralArrayType(Context.CharTy, EnumName.size());
+    StringLiteral *EnumStr =
+        StringLiteral::Create(Context, EnumName, StringLiteralKind::Ordinary,
+                              /*Pascal*/ false, StrTy, OpLoc);
+    if (EnumStr->getString() == "") {
```

Comment:
This is not how you compare strings! 
```suggestion
    if (EnumStr->empty()) {
```